---
title: "Functional variation among bacterial taxa within host-associated microbiomes"
author: "Tobias Nunn"
format: html
editor: visual
---

# 1. Introduction

(500 words, 5-10 refs)

# 2. Research evidence

## 2.0. the reset / Mistakes

as important as the outputs of this project are to its legitimacy, me learning how to conduct my work in a proper scientific manner is as equally important of an outcome. This work started in the summer, when i had done nothing of the type before, so i had basically no idea what i was doing or what i was messing with. So i made mistakes, many mistakes. Then, i kept going, building on those mistakes until around christmas i realised i was completely out of my depth and i needed to reivaluate. My notetaking especially was crude by modern comparison. So i decided to do a reset, i believed this was important as it allowed me to start again from a more informed point, and "do things right this time". However, i did conduct analysis in the first half of the year that could be relevent, error prone as it may be. And i do not want my mistakes to be buried, they must be remembered so i do not repeat them, or so anyone else reading this does not repeat them. So i would like to dedicate section 2.0. to the mistakes i made and how i learned from them to gain skills and become a better data scientist, as i believe this step was the catalyst for all hereafter steps to be as good as they were, without these errors, i doubt i would have worked as hard to make the work i did after as good, and thats alright.

here::here for good pathing

proper integration of images into the notebooks

(i understand if you need to dock points for this, but i think it is too important to not write down)

## 2.0.2 summer stuff (?)

## 2.1.1 Check M/2

## 2.1.2 CheckM tree_qa (hmm, if i do it this way then imma need to make all these three hashes so i can have a 2.1 to intro)

## 2.2. Innitial small analysis

## 2.2.2 DIAMOND_RESULTS.tsv

## 2.3. GTDB-TK de_novo_wf

## 2.4. flye output analysis

## 2.5. Phylogenetic trees

## 2.6. Wolbachia side-project

### Intro

In mid-August 2024 I helped with my supervisor Aaron Comeault's paper ("Phylogenetic and functional diversity among Drosophila-associated metagenome-assembled genomes") as a learning exercise for EggNOG-mapper and heatmapping. After receiving notes from the reviewers, Aaron asked me to process two samples of note from the study, that were Wolbachia species through Eggnog mapper and GTDB-TK so as to identify specific "Cif" and "WOPhage" genes. Specifically, I was tasked with answering the questions:

-   Do they harbor the characterized Cif genes? or WOPhage?

-   What other Wolbachia are they closely affiliated with?

### Methods

Both of these processes were run using modified bash scripts I had already made for my own project on Hawk. Specifically; metagenomerunner.sh, metagenomequared.sh and wolbachia_gtdbtk.sh.

#### EggNOG-mapper

The first two scripts work in tandem, as their names suggest. The specifications for the eggnog analysis are set in metagenomerunner.sh, including the amount of memory to use, the input file location and where the output files should be located. metagenomequared.sh is passed a file containing a list of the file names and paths, I called this genera_control_file_batch_aa.tsv. metagenomesquared.sh uses this file to set parameters for the accession and filename based on the items in the list, then calls slurm to run metagenomerunner.sh, passing in the parameters. A while loop is used so that the rows can be itterated over, meaning I only need to run metagenomesquared.sh once to get both jobs loaded to hawk. Once the samples are annotated, a copy statement using command-line code at the end of metagenomerunner.sh copies the outputs to my home directory, where they are more easily transferred to my machine. Then, in R, i use a script called gene_analysis.R to load the outputs and attempt to find any reference to these "Cif" or "Wophage" genes. Unlike for the usual analysis, I do not run `Microbiomeprofiler::enrichKO()`, I am unsure whether enrichment would have been relevent for this analysis. Ultimately, none of these genes were found from this analysis.

#### blastn

In order to confirm or disprove the result above. Aaron told me to use a program called blastn. I ran this locally on the command line, my commands are stored in the script blaster.txt. These outputted the files; ztaro_tblastn_results.txt and ztsac_tblastn_results.txt

#### GTDB-TK

This process also began on hawk, using wolbachia_gtdbtk.sh. This process is simpler, as it takes a directory of .fasta files, for the un-annotated genomes. Then outputs a directory of files which I downloaded to my local machine. Then, using R, I read in the gtdbtk.bac120.decorated.tree output to the treemaker.R script, where I use the package ggtree to make the tree. However, Styling needed to be done for a more interpretable output, to aid this, I used the NCBI REST API to download metadata about the samples mentioned in the tree file, then used that to modify the tip labels and add identifiers of which samples were the local ones.

### Results

Do they harbor the characterized Cif genes? or WOPhage?

After sending the outputs of the blastn analysis (ztaro_tblastn_results.txt and ztsac_tblastn_results.txt) to Aaron. He determined that they do confirm the existence of these genes. As it was not directly for my project, I did not question how he confirmed this. That may have been a mistake on my part.

What other Wolbachia are they closely affiliated with?

```{r tree}
#| echo: FALSE
#| label: fig-tree
#| fig-align: center
#| fig-cap: "Phylogenetic tree showing which strains of Wolbachia are closest related to those isolated from Drosophila flies by Comeault et al."
knitr::include_graphics(here::here("../aaron_mini_proj/outputs/", "Wolbachia_tree.png"))
```

The tree presents that the closest strain to both of the bangor samples are \*Wolbachia pipentis\*.

### Scripts

this bit is by no means done, and i need to decide on how i want to do this bit, do i give a long list of individual files, directories? the other bits need refinement too, but that can wait until later.

## 2.7.1 EggNOG-mapper

## 2.7.2 Heatmaps (?) (they are heavily linked so it makes sense to do this i think)

## 2.8. API calls

## 2.9. Heatmaps

## 2.10. fun genomics papers

not \*really\* a direct part of the project, but learning to read the literature properly around ones areas of intrest, and doing so recreationally, is important for a professional scientist and something that i struggle with, so i think warrents a spot here, not so much as proof i have read, more so as a promise to follow through with actually doing it. It is a skill, i think, interpreting scientific text and focus more generally, so i think it is relevant enough to talk about, if not... thats ok, im going to do it anyway. Well... i have to read papers in order to reference this document, thus making paper reading a project-relevant skill to learn how to do better.

## 2.11. Other

i made API calls a section, maybe there are other sub-script level processes that warrent their own part

# 3. Conclusion / Discussion

(250 words 5-10 refs, this is split between analysis and skills)

![](images/clipboard-4027681411.png)

# 4. References
