---
author: "Tobias Nunn"
title: "Functional variation among bacterial taxa within host-associated microbiomes"
subtitle: "Supervisor: Aaron Comeault"
format: docx
editor: visual
bibliography: DISS_BIBLIOGRAPHY.bib
csl: harvard-university-of-leeds.csl
---

## Aims

Aim 1: determine whether bacterial genera living on or in different hosts have different genomic functional capacities.

Aim 2: determine whether different bacterial taxa (genera or families) differ in their predicted functional capacity independent of their host environment.

## Abstract

Bacterial Microbiomes are cryptic environments, in which bacteria exhibit cooperative and antagonistic interactions. Understanding these relationships could have consequences for many other fields, including health and conservation, due to the universal presence of bacteria and the functional roles they play in their environments. Due to the microscopic nature of these ecosystems, genomics is a useful way to study them, and much progress has been made on this in the field's 30 year history.

Publicly available genomic databases hold vast quantities of information that can assist in drawing new conclusions about the functional differences between taxa, specifically species and genus, and how their environment may shape these differences, and how this may impact potential hosts. This study shall find these differences by annotating many public bacterial genomes and using KEGG values to find functional differences. This will lead to a greater understanding of the functional roles bacteria play in these environments.

## Introduction

Bacterial genomics is an interesting field and there has been much evolution over the 30 years the field has existed. The first bacterial genome to be sequenced was that of *H. influenzae* in 1995 using shotgun sequencing [@fleischmann_whole-genome_1995]. Since then there have been many technological improvements, allowing genomes to be sampled from over 50 bacterial phyla, with continually more data being reported about lesser researched taxonomic groups [@land_insights_2015]. Much research has been done into the relationships between bacteria and host. For example, nutrient acquisition, the digestion of complex compounds, such as plant polysaccharides and the production of amino acids and vitamins, useful for the host. Often, these are functions that hosts cannot perform for themselves [@mendes_cross-kingdom_2015]. A greater understanding of this can have implications for many fields, such as: bioforensics; disease treatment; biosurveillance and metabolic modeling [@land_insights_2015]. There have been continually new generations of sequencing technology. The most modern and powerful is Oxford Nanopore [@abdi_revolutionizing_2024]. Many studies have utilised these methods, however few discuss microbial genomics in the context of relationships and interactions between microbiomes and their environment or host, and the factors that affect these relationships. A study on the microbiome of a species of potato plant (*Solanum tuberosum*) discussed how certain species of bacteria can alleviate the impacts of drought on crop yield. Potatoes are an important food globally and ensuring food stability is vital for future development [@petrushin_potato_2024]. As climate change worsens, water scarcity will become more common in some regions, so an understanding of how to make them more plastic to this effect is important.

The human gut microbiome has been studied frequently, probably due to the relationship between it and public health. For example Crohn's disease is significantly linked to lower gut microbiome richness. Similarly, a relationship between microbial load and variation was found [@vandeputte_quantitative_2017]. An understanding of Microbiomes can have implications for conservation efforts. For example, in combatting Chytridiomycosis, an infectious disease causing amphibian population decline. Previous research found that differences in skin microbiome can impact on susceptibility to infection [@jimenez_amphibian_2017]. This is supported by Harris et al. (2009), who also found that the skin microbiome is an important factor in disease survival. There are other important elements to the skin of amphibians, for example respiration, that the microbiome affects, making a greater understanding of the relationships and functions therein vital [@harris_skin_2009]. Another study found that members of the same species showed similar microbiomes, whereas there were fewer similarities between different individuals from the same pond [@mckenzie_co-habiting_2012]. This might be evidence of co-evolution between host species and microbiome. The practical uses of bacteria in the context of sustainability are varied. A recent study outlined how bacteria are being used to recycle waste gases or biomass and produce products like ethanol, other researchers have found bacteria that break down certain plastics [@rappuoli_microbes_2025].

Relationships between bacteria have been studied previously, in the context of cooperation in biofilm formation. Both antagonism and cooperation were found in the bacterial species studied. The sample was taken from drinking water, six species were identified and 96 agar plates with different combinations of these were created. Previous studies had found that cooperation lead to more stable communities and a few specific beneficial relationships were identified, for example, plasmid conjugation [@simoes_biofilm_2007]. A recent study by [@lunjani_mechanisms_2021] discussed host-microbe interactions in human skin with relevance to how modern factors are changing these relationships and causing inflammatory skin diseases. The study outlines how this knowledge could help in creating personalised probiotic solutions to this problem. There are many ways in which bacteria in a microbiome can interact, as mentioned previously, they can cooperate and produce biofilms and share nutrients, however, there also exists competition wherever there is a niche to exploit and limited resources. These relationships cause selection for genes that cause, for example, the production of antimicrobial compounds to inhibit the competitors.

There are many species of bacteria, as such there are almost limitless ways in which they interact with each other in different conditions, gaining better insight into some of the ways in which these relationships occur on a genetic level could have impacts for fields as broad as public health to pollution to zoonotic disease and conservation. This study will aim to investigate some of these relationships through bottom-up estimates of function based on gene content, using publically available data, and present these in the context of how they might have a beneficial impact on potential host organisms. This work may be cause for future research to be done in the area.

## Methods

This project is desk-based, and all data analysis will be undertaken computationally. Thus, data must be collected from online sources. The online genome data repository used by this project is the National Center for Biotechnology Information (NCBI) [genome database](https://www.ncbi.nlm.nih.gov/datasets/genome/). There are roughly 2.61 million bacterial genomes in this database that could be annotated. Along with this exists a range of metadata from taxonomy information to host and location data. This provides a benefit as the time to pull down this data is much shorter than the time taken to harvest samples from the field, then grow and sequence them in the lab. With this I can create novel analyses and draw new conclusions about how the evolutionary history shapes the relationships between bacteria. Another important database is the Kyoto Encyclopedia of Genes and Genomes (KEGG) [database](https://www.genome.jp/kegg/). This is important as it holds data on the function of genes, these are grouped into "Orthologs" ([KO values](https://www.genome.jp/kegg/ko.html)) of shared function. The database also has data on how these KOs fit into larger biological systems, such as metabolism. These are called "[pathways](https://www.genome.jp/kegg/pathway.html)". With this information, i can directly correlate the genes possessed by different species of bacteria to their potential functions in the bacteria and thus, draw conclusions about how those functions could shape relationships with potential hosts and other bacteria in the environment. Specifically, grouped analysis comparing genera will be done to visualise how the evolutionary differences between genera can cause differential genome content and gene function.

In order to compare the genomes against the KEGG database, they must first be transformed. Firstly, [EggNOG-mapper/2.1.12](https://github.com/eggnogdb/eggnog-mapper/blob/master/README.md) will be utilised to annotate the genomes using slurm scripting, this will identify the genes of the genome. After this, the R package [MicrobiomeProfiler](https://bioconductor.org/packages/release/bioc/html/MicrobiomeProfiler.html) will be used, specifically the `EnrichKO()` function, which will [enrich](https://rdrr.io/github/YuLab-SMU/MicrobiomeProfiler/man/enrichKO.html) the annotated genomes so as to identify the significant KEGG pathways, their functions can be discovered through comparison with the database. From this point, visualisations such as heatmaps will be used to present these pathways. In order to accomplish this, groupings will need to be created, such as genera inside a specific family of bacteria or host taxonomy. These can then be analysed in the context of the evolutionary history or environment of the groups selected. Using this wealth of information. A secondary item of this project is to explore alternatives, microbial genomics is a cutting edge field, with more tools emerging continuously, and finding which systems or processes work best for these analysis will be important for continued research. For example, the enrichment tool [KofamKOALA](https://www.genome.jp/tools/kofamkoala/) or the [gene ontology (GO)](https://geneontology.org/) system have been identified as possible alternatives, and can be compared with the current methods to see which provides the best data clarity, accuracy and rapidity.

Due to the size of the dataset, efficient storage of this data will be vital so that the computational systems do not get overwhelmed. To aid this, a relational database will be created in PostgreSQL. PostgreSQL is good at handling the .JSON data format which is used in many public APIs, including the NCBI. A relational database will allow questions about the data to be answered by querying; host information, taxonomy, etc. For the purposes of data transparency and experiment reproducibility, the scripts used in this project will be uploaded to a github repo for the project. So that any researchers who wish to build on this project can take inspiration from or adapt the scripts used. This repo will also contain a series of notebooks outlining where the data came from, the specifics of the data pipeline and the significance of the outputs. API calls are efficient ways of transferring large sums of information, which will allow for more complex analysis to be performed. Due to the lack of manual steps in this, the code should provide clarity for future researchers to more easily reproduce or understand the step.

Finally, the data will be supplemented by over 100 genomes generated in the labs at ECW by Bangor Masters and PhD researchers. These samples come from amphibian skin microbiomes. The known status of these samples mean that more exact analysis can be performed, as well as providing good comparisons. For example, there are not many samples from amphibian skin microbiomes in the database. Thus, a comparison of how the same genera differ when on these hosts compared with the environment could be created. This will lead to other types of visualisation to be created, for example, phylogenetic trees to confirm the taxonomic assignment of these samples.

## References
